components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -1511,6 +1511,13 @@ public Response deleteAPI(String apiId, String ifMatch, MessageContext messageCo
                 RestApiUtil.handleConflict("Cannot remove the API because following resource paths " +
                         usedProductResources.toString() + " are used by one or more API Products", log);
             }
+
+            // check user has publisher role and API is in published or deprecated state
+            if (!APIUtil.isRoleExistForUser(username, "Internal/publisher") && (
+                    APIConstants.PUBLISHED.equalsIgnoreCase(api.getStatus()) || APIConstants.DEPRECATED
+                            .equalsIgnoreCase(api.getStatus()))) {
+                RestApiUtil.handleConflict(username + " cannot remove the API", log);
+            }
```

Comment:
We cannot use "Internal/publisher" role always to check for a publisher role. This can be changed from scope assignments.

We need to check this using scopes. Need to check user has "apim:api_publish" scope instead of `APIUtil.isRoleExistForUser(username, "Internal/publisher")`

Scopes of the user can be found using:
```
            String[] tokenScopes =
                    (String[]) PhaseInterceptorChain.getCurrentMessage().getExchange()
                            .get(RestApiConstants.USER_REST_API_SCOPES);
```